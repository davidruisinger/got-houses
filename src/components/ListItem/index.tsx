import React, { FunctionComponent } from 'react'
import posed from 'react-pose'
import styled from '../_utils/styledComponents'
import Card from '../Card'
import FlexBox from '../FlexBox'

const animationProps = {
  hoverable: true,
  pressable: true,
  init: {
    scale: 1,
  },
  hover: {
    scale: 1.1,
  },
  press: {
    scale: 0.95,
  },
}

const AnimatedWrapper = styled(posed(Card)(animationProps))`
  padding: 0.5em;
  box-sizing: border-box;
  cursor: pointer;
  background: ${props => props.theme.canvas_first};
  z-index: 9;

  @media (hover: hover) {
    :hover {
      z-index: 999999;
    }
  }
`

const Inner = styled(FlexBox)`
  width: 100%;
  height: 100%;
`

const Name = styled.h3`
  text-align: center;
  font-size: 1.1em;
`

interface Props {
  label: string
  onClick: (id: string) => void
}

const ListItem: FunctionComponent<Props> = ({ label, onClick }) => (
  <AnimatedWrapper onClick={onClick}>
    <Inner justifyContent="center" alignItems="center">
      <Name>{label}</Name>
    </Inner>
  </AnimatedWrapper>
)

export default ListItem
