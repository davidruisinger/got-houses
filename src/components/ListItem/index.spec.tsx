import 'jest-styled-components'
import React from 'react'
import { shallow, mount } from 'enzyme'
import ListItem from './index'

describe('<ListItem />', () => {
  const label = 'Test label'
  const mockClick = jest.fn()
  const shallowedWraper = shallow(
    <ListItem label={label} onClick={mockClick} />
  )
  const mountedWrapper = mount(<ListItem label={label} onClick={mockClick} />)

  it('matches the snapshot', () => {
    expect(shallowedWraper).toMatchSnapshot()
  })

  it('renders the label', () => {
    expect(
      mountedWrapper
        .find('h3')
        .first()
        .text()
    ).toEqual(label)
  })

  it('handles clicks', () => {
    shallowedWraper.simulate('click')
    expect(mockClick).toHaveBeenCalledTimes(1)
  })
})
