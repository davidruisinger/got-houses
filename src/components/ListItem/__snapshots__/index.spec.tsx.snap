// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItem /> matches the snapshot 1`] = `
<Styled(Component)
  onClick={[MockFunction]}
>
  <Styled(styled.div)
    alignContent="stretch"
    alignItems="center"
    direction="row"
    justifyContent="center"
    wrap="nowrap"
  >
    <styled.h3>
      Test label
    </styled.h3>
  </Styled(styled.div)>
</Styled(Component)>
`;
